applications/GeoMechanicsApplication/custom_conditions/T_condition.h

```diff
@@ -78,6 +66,15 @@ class KRATOS_API(GEO_MECHANICS_APPLICATION) GeoTCondition : public Condition {
                               const ProcessInfo& rCurrentProcessInfo);
 
 private:
+    [[nodiscard]] DofsVectorType GetDofs() const
+    {
+        auto       result = DofsVectorType{};
+        const auto nodes  = this->GetGeometry();
+        std::transform(nodes.begin(), nodes.end(), std::back_inserter(result),
+                       [](const auto& r_node) { return r_node.pGetDof(TEMPERATURE); });
+        return result;
+    }
+
```

Comment:
Yes, we might be able to extract even more functionality here. However, I would propose to do that after we have rearranged the DOFs of the U-Pw elements, since that will make things simpler.

Also, I did move the implementation from the header file to the implementation file, as you suggested.